spec/lib/page_tracker_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe PageTracker do
+  let(:session) { {} }
+  let(:trainee) { build(:trainee) }
+  let(:path_a) { "/trainees/#{trainee.slug}/review-draft" }
+  let(:path_b) { "/trainees/#{trainee.slug}/personal-details/confirm" }
+  let(:path_c) { "/trainees/#{trainee.slug}/personal-details/edit" }
+  let(:history_sesssion_key) { "history_for_#{trainee.slug}" }
+  let(:origin_pages_sesssion_key) { "origin_pages_for_#{trainee.slug}" }
+
+  describe "#save!" do
+    context "within trainee specific pages" do
+      it "only saves GET requests" do
+        request = double(fullpath: path_a, head?: false, get?: false, patch?: true, put?: false)
+
+        PageTracker.new(trainee.slug, session, request).save!
+
+        expect(session[history_sesssion_key]).to be_empty
+      end
+
+      it "stores the pages in the order they were requested" do
+        request_a = double(fullpath: path_a, head?: false, get?: true, patch?: false, put?: false)
```

Comment:
Could we structure this spec in an arrange/act/assert structure. Everything is placed within the it blocks, I think they should just do simple assertions to the make tests a little more readable.

Reply:
I don't normally go for this approach, but in this particular case I felt it would be easier to read. I think the specs are long already and I was worried if I start putting in more `context` and `before` blocks, the it just gets much harder to follow.

However, I've tried what you suggested and it's not as bad as I thought, so all good!